src/Http/BaseApplication.php

```diff
@@ -110,6 +118,10 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
      */
     protected function getDispatcher()
     {
-        return new ActionDispatcher(null, null, DispatcherFactory::filters());
+        if ($this->dispatcher === null) {
```

Comment:
@markstory 

On the other thought, I don't think we can instantiate `ActionDispatcher` in a constructor because of the `DispatcherFactory::filters()` dependency. FIlters are usually added in a `bootstrap.php` file so after the application object has been created, am I correct?